read_wrapper.c

```diff
@@ -58,8 +60,32 @@ ssize_t read_wrapper(struct input_plugin_data *ip_data, void *buffer, size_t cou
 	}
 	if (count + ip_data->counter > ip_data->metaint)
 		count = ip_data->metaint - ip_data->counter;
-	rc = read(ip_data->fd, buffer, count);
+	rc = read(conn, buffer, count);
 	if (rc > 0)
 		ip_data->counter += rc;
 	return rc;
 }
+
+ /* duplicated from file.c to avoid adding openssl to cmus-remote dependencies */
+ssize_t read_all_from_conn(struct connection *conn, void *buf, size_t count)
+{
+	char *buffer = buf;
+	ssize_t pos = 0;
+
+	do {
+		ssize_t rc;
+
+		rc = conn->read(conn, buffer + pos, count - pos);
+		if (rc == -1) {
+			if (errno == EINTR || errno == EAGAIN)
```

Comment:
ssl bio can not also return other types of return codes that indicate `try again` and/or `interrupted` or `oef`.